actnow/accounts/models.py

```diff
@@ -18,6 +18,9 @@ def create_user(self, email, password=None, **extra_fields):
             raise ValueError("Users must have an email address")
 
         email = self.normalize_email(email)
+        # Since Users have to login in order to complete OAuth.
+        # They will have no permission to see anything in the admin dashboard.
+        extra_fields.setdefault("is_staff", True)
```

Comment:
1. Can you clarify what does the comment mean?
2. Can we customize the login templates for OAuth?

Reply:
Was just trying to justify why we need to give the users `is_staff` permission. But to clarify, setting a user as `is_staff` only allows him/her to login to the admin dashboard but not perform any further actions other than changing their password.
Lemme find out if we can >Can we customize the login templates for OAuth? 
and if we can, we won't(I think) need to set the user to `is_staff`

Reply:
What did we find about customizing templates?

Reply:
>Can we customize the login templates for OAuth?

OAuth doesn't have/require a login page. What it needs is just the `request.user` to be set to a logged-in User. 
So, we can create a view for `OAuth` that uses the custom template and then change [this](https://github.com/CodeForAfrica/actNOW/blob/main/actnow/settings.py#L57) to point to the new URL.

Reply:
@kilemensi  
I've created the following login page and it works although it now introduces new packages e.g `crispy_forms`.   
![Screenshot 2021-06-21 at 13 40 14](https://user-images.githubusercontent.com/13068580/122749648-40431e80-d296-11eb-9dd8-6f1e2ac1f720.png)

Reply:
For my understanding at least, what package requiries `crispy_forms`? I'm also looning at DRF docs and they seem to be using [Bootstrap 3](https://www.django-rest-framework.org/topics/html-and-forms/#using-template-packs).

PS: For Oauth we need to this about these [templates](https://django-oauth-toolkit.readthedocs.io/en/latest/templates.html). Especially the `authorize.html `template that will show the user the list of scopes requested and options to access / reject the request.

Reply:
Now I'm a bit lost @kilemensi 
When you said 
>Can we customize the login templates for OAuth?
Weren’t you referring to changing from the default Django admin site login to a `custom` login page?

Reply:
_Oauth_ @esirK ... I'm only interested in Oauth.

For example, when you click login with GitHub, you're presented with a login screen from GitHub and once you're logged in, you're presented with a list of requested scopes for you to allow or deny.

What/where are these templates and can we customize them?

Reply:
>For example, when you click login with GitHub, you're presented with a login screen from GitHub 

I see. This is what I described [here](https://github.com/CodeForAfrica/actNOW/pull/37/files#r655209960)

So in addition to changing the login page, we need to change the default page showing the scopes? If so, how should they look like?

Reply:
I've seen it's possible to edit the pages by just creating the templates inside `templates/oauth2_provider`. I've included `authorize.html` just to direct/show where the edits will be made.

Reply:
> > For my understanding at least, what package requires crispy_forms?

Not really a package but we need it to help us with the rendering of the forms instead of manually rendering the forms.

Reply:
If not required by the core dependencies (django, DRF, OauthTookit), then lets leave it out for now since we only need to the most basic of customisations.